public/javascripts/data/paragraph.js

```diff
@@ -2,19 +2,26 @@ const Database = require('better-sqlite3');
 
 
 // Function writes paragraph to the database
-exports.insertParagraph = (paragraph) => {
+exports.insertParagraph = (paragraph, res, req) => {
 
     const db = new Database('public/javascripts/data/stories.db');
     // TODO:
-    // 1. (user_id, stories_id) should contain current user and current story
+    // 1. (stories_id) should contain current story
     
+    // Get current user_id
+    const currentUserName = req.locals.userName;
+    const userIdStmt = db.prepare('SELECT id FROM users WHERE user_name = (?)');
+    const userIdGet = userIdStmt.get(currentUserName);
+    const userId = userIdGet.id;
```

Comment:
These 4 lines (11 - 15) can be replaced by setting the user ID onto the `req.locals` object.

In addition, you want to pass the `userId` as a parameter to this method, and do not rely on `req` and `res` in the data layer

Reply:
I'm having a hard time understand the difference between using `req.session` and `req.locals` . From what I've read `res.locals` can be passed to the view layer (.pug).

But in `controllers/login-out.js` I have set these properties to `req.session` for a POST request to login :
```
req.session.userId = userId;
req.session.isAuth = true;
req.session.userName = username;
```
Where I'm not clear is, what is the difference between setting these properties like I did above as opposed to setting them to `req.locals` ?